### ISSUE TITLE
Incorrect padding

### ISSUE THREAD

**Issue by isotoxin** [TokTok/c-toxcore#311]
https://github.com/TokTok/c-toxcore/blob/master/toxcore/Messenger.c#L2666

Guys, why your tests didn't failed?

Lets see my fix instead of https://github.com/TokTok/c-toxcore/pull/281

```
#pragma pack(push, 8) // just add this before SAVED_FRIEND
struct SAVED_FRIEND {
.......
};
#pragma pack(pop) // and just add this after SAVED_FRIEND
```

**Comment by iphydf** [comment]
Your fix is not standard C.

We changed the code to use the layout on AMD64 Linux (LP64). What data model does Windows use? Which failures do you see? I expect the assert to fail, and we will remove that assert today.

**Comment by isotoxin** [comment]
> Your fix is not standard C.
But gcc supports it.
Just see https://github.com/zetok/tox-spec/blob/master/spec.md
Where pudding must be? Right, after Status message (due it has size 1007 bytes), not after name

**Comment by iphydf** [comment]
You're right, the code seems wrong. We'll add a test that fails, and then fix it.

**Comment by iphydf** [comment]
Fixed in #313.
